src/main/java/dan200/computercraft/shared/peripheral/diskdrive/DiskDrivePeripheral.java

```diff
@@ -194,6 +197,46 @@ public final void ejectDisk()
         return disk.getItem() instanceof ItemDisk ? new Object[] { ItemDisk.getDiskID( disk ) } : null;
     }
 
+    /**
+     * Returns the type of media currently in the disk drive.
+     *
+     * @return The type of media currently in the disk drive.
+     * @cc.treturn string|nil The type of media currently in the disk drive, or {@code nil} if the type could not be determined.
+     */
+    @LuaFunction
+    public final String getDiskType()
+    {
+        ItemStack stack = diskDrive.getDiskStack();
+        if ( stack.isEmpty() )
+        {
+            return "empty";
+        }
+        else if ( stack.getItem() instanceof ItemDisk )
+        {
+            return "disk";
+        }
+        else if ( stack.getItem() instanceof ItemComputerBase )
+        {
+            return "computer";
+        }
+        else if ( stack.getItem() instanceof ItemTurtle )
+        {
+            return "turtle";
+        }
+        else if ( stack.getItem() instanceof ItemPocketComputer )
+        {
+            return "pocket";
+        }
+        else if ( MediaProviders.get( stack ) != null && MediaProviders.get( stack ).getAudio( stack ) != null )
+        {
+            return "record";
+        }
+        else
+        {
+            return null;
+        }
```

Comment:
This should really be a (default) method on `IMedia` (and then overridden on the various `IMedia` implementations) rather than an if-else block here. Probably worth doing this the proper OOP (and more importantly, extensible) way.